hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -675,10 +690,11 @@ public BootstrapStateHandler.Lock lock()
 
     @Override
     public void unlock() {
-      snapshotDeletingService.getBootstrapStateLock().unlock();
-      rocksDbCheckpointDiffer.getBootstrapStateLock().unlock();
-      sstFilteringService.getBootstrapStateLock().unlock();
-      keyDeletingService.getBootstrapStateLock().unlock();
+      snapshotDeletingService.ifPresent(deletingService -> deletingService.getBootstrapStateLock().unlock());
+      rocksDbCheckpointDiffer.ifPresent(
+          rocksDBCheckpointDiffer -> rocksDBCheckpointDiffer.getBootstrapStateLock().unlock());
+      sstFilteringService.ifPresent(filteringService -> filteringService.getBootstrapStateLock().unlock());
+      keyDeletingService.ifPresent(deletingService -> deletingService.getBootstrapStateLock().unlock());
```

Comment:
```suggestion
      locks.forEach(BootstrapStateHandler.Lock::unlock);
```

Note: this will release locks in the same order as they were acquired, instead of in reverse order.  I think both are fine as long as the order is fixed.

Reply:
Updated PTAL.

> Note: this will release locks in the same order as they were acquired, instead of in reverse order. I think both are fine as long as the order is fixed.

I think it should be fine.

However, seems like  `lock` and `unlock` call are not synchronized properly on the `Lock` object. This might cause possible deadlocks. Shall we synchronize on the lock object (`synhronized(this)`) inside the `lock` and `unlock` call?

Reply:
Do you mean the `lock`/`unlock` methods of `OMDBCheckpointServlet.Lock`?